### DISCUSSION TITLE
FastApi Construct generates Multiple Content Length in response creating client confusion

### DISCUSSION THREAD

**Discussion by bheuse** [fastapi/fastapi#10806] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/configurationitems", tags=["ConfigurationItem"])  # , response_model=ConfigurationItem
def post_ConfigurationItem(new_item : ConfigurationItem, params : ParamsList = Depends(), request: Request = None, response: Response = None):
    print("post_ConfigurationItem")
    return { "TT" : "DD" }
```

### Description

Multiple Content Length in response creating client confusion 

```
@app.post("/configurationitems", tags=["ConfigurationItem"])  # , response_model=ConfigurationItem
def post_ConfigurationItem(new_item : ConfigurationItem, params : ParamsList = Depends(), request: Request = None, response: Response = None):
    print("post_ConfigurationItem")
    return { "TT" : "DD" } 

```
Having a Response object and a return dict, causes the HTTP Response to contain 2 content-lenght headers and confuses the client - as this is not in line with the HTTP protocol.

Some error message could be raised when such construct are used - it took me 2-3 days to figure out the issue.

### Operating System

Windows

### Operating System Details

Windows

### FastAPI Version

Latest

### Pydantic Version

Latest

### Python Version

3.11

### Additional Context

Multiple Content Length in response creating client confusion 

```
@app.post("/configurationitems", tags=["ConfigurationItem"])  # , response_model=ConfigurationItem
def post_ConfigurationItem(new_item : ConfigurationItem, params : ParamsList = Depends(), request: Request = None, response: Response = None):
    print("post_ConfigurationItem")
    return { "TT" : "DD" } 

```
Having a Response object and a return dict, causes the HTTP Response to contain 2 content-lenght headers and confuses the client - as this is not in line with the HTTP protocol.

Some error message could be raised when such construct are used - it took me 2-3 days to figure out the issue.

**Comment by Kludex** [comment, marked as answer]
Please present a [MRE](https://stackoverflow.com/help/minimal-reproducible-example).
